tests/mtouch/MTouch.cs

```diff
@@ -3427,6 +3427,46 @@ public void NoProductAssemblyReference (bool nofastsim)
 			}
 		}
 
+		[Test]
+		[TestCase (true)]
+		[TestCase (false)]
+		public void MixedModeAssembliesCanNotBeLinked (bool nofastsim)
+		{
+			foreach (var flavor in new Tuple<MTouchLinker, bool> [] {
+					new Tuple <MTouchLinker, bool> (MTouchLinker.DontLink, false),
+					new Tuple <MTouchLinker, bool> (MTouchLinker.LinkAll, true),
+					new Tuple <MTouchLinker, bool> (MTouchLinker.LinkSdk, true) }) {
+
+				using (var mtouch = new MTouchTool ()) {
+					var tmp = mtouch.CreateTemporaryDirectory ();
+					string libraryPath = Path.Combine (tmp, "MixedClassLibrary.dll");
+					File.Copy (Path.Combine (Configuration.SourceRoot, "tests/common/MixedClassLibrary.dll"), Path.Combine (tmp, "MixedClassLibrary.dll"));
+
+					mtouch.CreateTemporaryApp (code: "public class TestApp { static void Main () { System.Console.WriteLine (typeof (MixedClassLibrary.Class1)); System.Console.WriteLine (typeof (ObjCRuntime.Runtime)); } }",
+											   extraArg: $"-r:Xamarin.iOS.dll -r:{StringUtils.Quote (libraryPath)}");
```

Comment:
Can't you just reference the committed `MixedClassLibrary.dll` instead of copying it first?

```csharp
-r:{StringUtils.Quote (Path.Combine (Configuration.SourceRoot, "tests", "common", "MixedClassLibrary.dll"))}
```